workflow-manager/bucket/bucket.go

```diff
@@ -189,7 +192,9 @@ func (b *Bucket) gcsClient() (*storage.Client, error) {
 }
 
 func (b *Bucket) listFilesGS() ([]string, error) {
-	ctx, cancel := utils.ContextWithTimeout()
+	// This timeout has to cover potentially numerous roundtrips to the
+	// paginated API for listing objects, so we use a longer timeout than usual.
+	ctx, cancel := context.WithTimeout(context.Background(), 120*time.Second)
```

Comment:
This is a critical part of the application right? Does having *any* timeout even make sense here? Maybe it's a sign that something is wrong if it fails, but if the application can't reasonably recover from this timeout being cancelled would it make sense to make it even larger to handle variations in networking/downtime of gs? (e.g. something like 10 minutes)

Reply:
We already will retry 10 minutes later because `workflow-manager` will run again. The point of a timeout here is so that we don't hang forever if the GCS server just never replies but also never closes the connection: we'd rather fail obviously with an error than just have a particular run of the task never exit.

Reply:
I agree we can afford to make this higher. That said, I also think it's time to change how we list files so we list by a pattern that matches the things we're looking for, so we don't have to paginate through so much unwanted stuff.